### ISSUE TITLE
Closing an InputStream twice causes subsequent reads to fail with a NoSuchFileException

### ISSUE THREAD

**Issue by oxbowlakes** [marschall/memoryfilesystem#49]
This code fails:

  ```java
    import java.nio.file.*; import java.io.*;
    @Test
    public void testMemoryFileSystemMultipleClose() {
      FileSystem fs = MemoryFileSystemBuilder.newLinux().build("ID");
      Path foo = fs.getPath("foo");

      Files.write(foo, java.util.Collections.singleton("Foo"));

      InputStream is = Files.newInputStream(foo);
      is.close();
      is.close(); //remove the second call to close() and all is OK

      InputStream is2 = Files.newInputStream(foo); //NoSuchFileException
    }
```

This is because the `openCount` of a `MemoryFile` being less than zero is an indication that the file does not exist. I would suggest one of two fixes.

First option: `BlockInputStream` keeps track of whether it has been closed; closing an already-closed stream is a no-op. This seems to be the behaviour of the standard library's streams (as I have found this "bug" in code which works in reality!)

```java
    @Override
    public void close() throws IOException {
      if (this.checker.isOpen()) {
        this.checker.close();
        this.memoryContents.modified();
        this.memoryContents.closedStream(this.path, this.deleteOnClose);
      }
    }
```

Second option: closing a stream/channel on a Path should not let the openCount drop below zero. that is

```java
    this.openCount = Math.max(this.openCount - 1, 0)
```
However, I have not tested either changes

**Comment by marschall** [comment]
The documentation specifies that #close on an already closed stream is a no-op. I'll check if the OutputStream and the ByteChannels have the same issue.

**Comment by marschall** [comment]
BlockOutputStream had the same issue. BlockChannel didn't have the issue since it extends AbstractInterruptibleChannel.

**Comment by marschall** [comment]
I just released 0.6.6 that should fix the issue. It may take a few hours until it shows up in Maven Central.
